app/javascript/src/clone.ts

```diff
@@ -0,0 +1,3 @@
+export function clone(obj: any): any {
```

Comment:
I wouldn't use any here.  Instead use a generic.

```suggestion
export function clone<T>(obj: <T>): T {
```

I think since JSON.parse returns an any, it will be coerced by default.